light/txpool.go

```diff
@@ -365,6 +367,21 @@ func (pool *TxPool) validateTx(ctx context.Context, tx *types.Transaction) error
 		return core.ErrNonceTooLow
 	}
 
+	// Ensure gold transfers are whitelisted if transfers are frozen.
```

Comment:
Looks like there's a bunch of duplicated code here, can we share this some how?

Reply:
Good thought